openwisp_monitoring/check/classes/iperf.py

```diff
@@ -0,0 +1,205 @@
+        """
+        Executes device command
+        """
+        return dc.connector_instance.exec_command(command, raise_unexpected_exit=False)
+
+    def _connect(self, dc):
+        """
+        Connects device returns its working status
+        """
+        return dc.connect()
+
+    def _get_iperf_result(self, res, exit_code, device, mode):
+        """
+        Get iperf test result
+        """
+
+        res_dict = json.loads(res)
+        if mode == 'TCP':
+            if exit_code != 0:
+                logger.warning(
+                    f'Iperf check failed for "{device}", {res_dict["error"]}'
+                )
+                return {
+                    'iperf_result': 0,
+                    'sent_bps_tcp': 0.0,
+                    'received_bps_tcp': 0.0,
+                    'sent_bytes_tcp': 0.0,
+                    'received_bytes_tcp': 0.0,
+                    'retransmits': 0,
+                }
+            else:
+                # Gbps = Gigabits per second
+                # GB = GigaBytes
+                # Todo : Remove below coversion once
+                # https://github.com/openwisp/openwisp-monitoring/pull/397 get merged
+                sent_json = res_dict['end']['sum_sent']
+                recv_json = res_dict['end']['sum_received']
+                sent_bytes = sent_json['bytes']
+                sent_bytes_GB = sent_bytes / 1000000000
+                sent_bps = sent_json['bits_per_second']
+                sent_Gbps = sent_bps / 1000000000
+                received_bytes = recv_json['bytes']
+                received_bytes_GB = received_bytes / 1000000000
+                received_bps = recv_json['bits_per_second']
+                received_Gbps = received_bps / 1000000000
+                retransmits = sent_json['retransmits']
+                return {
+                    'iperf_result': 1,
+                    'sent_bps_tcp': round(sent_Gbps, 2),
+                    'received_bps_tcp': round(received_Gbps, 2),
+                    'sent_bytes_tcp': round(sent_bytes_GB, 2),
+                    'received_bytes_tcp': round(received_bytes_GB, 2),
+                    'retransmits': retransmits,
+                }
+
+        elif mode == 'UDP':
+            if exit_code != 0:
+                logger.warning(
+                    f'Iperf check failed for "{device}", {res_dict["error"]}'
+                )
+                return {
+                    'iperf_result': 0,
+                    'sent_bps_udp': 0.0,
+                    'sent_bytes_udp': 0.0,
+                    'jitter': 0.0,
+                    'total_packets': 0,
+                    'lost_packets': 0,
+                    'lost_percent': 0.0,
+                }
+            else:
+                sent_bytes = res_dict['end']['sum']['bytes']
+                sent_bytes_MB = sent_bytes / 1000000
+                sent_bps = res_dict['end']['sum']['bytes']
+                sent_Mbps = sent_bps / 1000000
```

Comment:
Why not in Gbps?

Reply:
It always comes out very small value when I changed to `Gbps`. We'll remove these manual conversions once https://github.com/openwisp/openwisp-monitoring/pull/397 get merged :+1: